tracing/lightstep_tracer.go

```diff
@@ -0,0 +1,65 @@
+package tracing
+
+import (
+	"context"
+	"github.com/lightstep/otel-launcher-go/launcher"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type Tracer interface {
+	Trace(commandName string) Span
+	Shutdown()
+}
+
+type Span interface {
+	End()
+}
+
+type LightstepTracer struct {
+	launcher launcher.Launcher
+	tracer   trace.Tracer
+	ctx      context.Context
+	spanID   string
+}
+
+func NewLightstepTracer(service, token string) *LightstepTracer {
+	return &LightstepTracer{
+		launcher: launcher.ConfigureOpentelemetry(
+			launcher.WithServiceName(service),
+			launcher.WithAccessToken(token),
+		),
+		tracer: otel.Tracer(service),
+	}
+}
+
+func (t *LightstepTracer) Trace(name string) Span {
+	if t.ctx == nil {
+		t.ctx = context.Background()
+	}
+
+	var span trace.Span
+
+	if t.spanID == "" {
+		t.ctx, span = t.tracer.Start(t.ctx, name)
+		t.spanID = span.SpanContext().SpanID().String()
+	} else {
+		_, span = t.tracer.Start(t.ctx, name)
+	}
```

Comment:
I'm having trouble seeing the motivation behind this. Are we trying to protect the developer from calling `Trace(name string) Span` twice?

Reply:
This is a hack. `t.ctx` must only be overwritten if there is no `spanID`. If you overwrite the context while it has a  `spanID` the next span appears as a child of the previous span. So if you do that on every iteration you will see something like a linkedlist of spans. We want more of a trie of spans where the first span is the parent and all other spans are direct children of the root.

Reply:
I'm not sure how to solve this issue without setting the context on the struct. Through the context the span knows which parent trace it has. If we have multiple http calls that are part of a parent process (such as parsing a block), we need to pass that parent trace along to all the spans. 

Alternatively, we could recreate a context for every http call. So we store individual context values on the struct and make a new context based on those values.